python/pyarrow/_s3fs.pyx

```diff
@@ -237,11 +237,20 @@ cdef class S3FileSystem(FileSystem):
                                         'port': 8020, 'username': 'username',
                                         'password': 'password'})
     allow_bucket_creation : bool, default False
-        Whether to allow CreateDir at the bucket-level. This option may also be
+        Whether to allow directory creation at the bucket-level. This option may also be
         passed in a URI query parameter.
     allow_bucket_deletion : bool, default False
-        Whether to allow DeleteDir at the bucket-level. This option may also be
+        Whether to allow directory deletion at the bucket-level. This option may also be
         passed in a URI query parameter.
+    check_directory_existence_before_creation : bool, default false
+        Whether to check the directory existence before creating it.
+        if false, when creating a directory the code will not check if it already
+        exists or not. It's an optimization to try directory creation and catch the error,
+          rather than issue two dependent I/O calls.
```

Comment:
```suggestion
        If false, when creating a directory the code will not check if it already
        exists or not. It's an optimization to try directory creation and catch the error,
        rather than issue two dependent I/O calls.
```